bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/handler/VeluxBridgeHandler.java

```diff
@@ -253,71 +258,85 @@ public boolean bridgeCommunicate(BridgeCommunicationProtocol communication) {
-        // Take care of unusual situations...
+        // take care of unusual situations...
         if (scheduler.isShutdown()) {
-            logger.warn("initialize(): scheduler is shutdown, aborting the initialization of this bridge.");
+            logger.warn("initialize(): scheduler is shutdown, aborting initialization.");
             return;
         }
-        getTaskExecutor();
-        logger.trace("initialize(): preparing background initialization task.");
-        // Background initialization...
         scheduler.execute(() -> {
-            logger.trace("initialize.scheduled(): Further work within scheduler.execute().");
-            logger.trace("initialize.scheduled(): Initializing bridge configuration parameters.");
-            this.veluxBridgeConfiguration = new VeluxBinding(getConfigAs(VeluxBridgeConfiguration.class)).checked();
-            logger.trace("initialize.scheduled(): work on updated bridge configuration parameters.");
-            bridgeParamsUpdated();
-
-            logger.debug("initialize.scheduled(): activated scheduler with {} milliseconds.",
-                    this.veluxBridgeConfiguration.refreshMSecs);
-            refreshJob = scheduler.scheduleWithFixedDelay(() -> {
-                try {
-                    refreshOpenHAB();
-                } catch (RuntimeException e) {
-                    logger.warn("Exception occurred during activated refresh scheduler: {}.", e.getMessage());
-                }
-            }, this.veluxBridgeConfiguration.refreshMSecs, this.veluxBridgeConfiguration.refreshMSecs,
-                    TimeUnit.MILLISECONDS);
-            logger.trace("initialize.scheduled(): done.");
+            initializeSchedulerJob();
         });
-        logger.trace("initialize() done.");
     }
 
     /**
-     * NOTE: It takes care about shutting down the connections before removal of this binding.
+     * Various initialisation actions to be executed on a background thread
+     *
+     * Note: if a disposeSchedulerJob() has been scheduled for the same ThingUID, then execution of this method
+     * will be paused until that job has finished.
      */
+    private synchronized void initializeSchedulerJob() {
+        logger.trace("initializeSchedulerJob(): initialize bridge configuration parameters.");
+        veluxBridgeConfiguration = new VeluxBinding(getConfigAs(VeluxBridgeConfiguration.class)).checked();
+        Future<Boolean> disposeJob = disposeSchedulerJobs.remove(veluxBridgeConfiguration.ipAddress);
+        if ((disposeJob != null) && (!disposeJob.isDone())) {
+            logger.trace("initializeSchedulerJob(): wait for pending disposeSchedulerJob() to finish.");
+            try {
+                disposeJob.get();
+            } catch (InterruptedException | ExecutionException e) {
+                logger.warn("initializeSchedulerJob(): unexpected exception waiting for disposeSchedulerJob() '{}'.",
+                        e.getMessage());
+            }
```

Comment:
The `InterruptedException` is thrown when OH is shut-down. That means the current operation should be finished as fast as possible and no further operations should be done. You can achieve this by not catching the exception here but throw the InterruptedException and catch it at the highest level you can. As this is not an error, it shouldn't be logged, but only handled.

Reply:
@fwolter thank you for this review. You comment raises an interesting question, that I hope you can advise on..

**The Bad News**:  As stated in the PR title, the purpose is to resolve issues in the Velux Bridge during initialization and shutdown. To be specific: if the connection from OH to the bridge is not shut down in a very deliberate manner, then the bridge will go into a 'zombie' state and will refuse reconnections when OH is restarted (requiring a power-cycle to fix it). i.e. the `dispose()` process must be very deliberate. So if OH would throw an InterruptedException during dispose(), then it would be voiding the whole point of this exercise !! => Ergo (IMHO) OH should NOT throw an InterruptedException at this point!!

**The Good News**:  In fact, during many tests of different forms of binding shut down, I have never actually observed InterruptedException being thrown at this point. => So perhaps this is not an issue anyway..

So I would appreciate your valuable thoughts on this matter before I make changes to the code.

Reply:
@fwolter I think there may be another problem with your suggestion: If we don't catch the InterruptedException at line 287 then initializeSchedulerJob() must throw it. But if initializeSchedulerJob() does throw it, then we must catch it at line 269 in initialize(). Because it is not allowed to schedule tasks in the thread-pool (via BaseThingHandler.scheduler.submit()) if the respective task does not handle all exceptions within itself. Or ??

Reply:
First thing I would do is raise a sh*t storm, that Velux fixes this bug :)

Just for clarification, the `InterruptedException` is thrown when the JVM is shut-down. E.g. when you call `systemctl stop openhab`.

When this exception is raised, you can do whatever is necessary to do a proper cleanup. But keep in mind that everything you do, does delay the execution of `systemctl stop openhab`.

Reply:
Ok. Thanks for the feedback. I have just made a new commit that (among other things) ensures that shutdown doesn't take longer than COMMUNICATION_TASK_MAX_WAIT_SECS = 10 seconds